apps/wappush/js/messagedb.js

```diff
@@ -230,7 +264,8 @@ var MessageDB = (function() {
   function mdb_retrieve(timestamp) {
     return mdb_open().then(function mdb_openResolved(db) {
       var state = { result: null };
-      var transaction = db.transaction(MESSAGE_STORE_NAME, 'readwrite');
+      var transaction =
+        db.transaction([ MESSAGE_STORE_NAME, VIEWED_STORE_NAME ], 'readwrite');
       var promise = mdb_transactionPromise(transaction, state);
```

Comment:
if we want to completely kill that `state` parameter (this is the last place where it's used), you can replace `return promise;` at the end of the function by `return promise.then(() => state);`